### PR TITLE
Nextcloud with disk and gateway

### PR DESCRIPTION
**Pull request by scottyeager**
This PR updates the Nextcloud solution to use a disk (for improved performance, especially with Docker) and also to support gateways.

### CODE REVIEW

File: packages/playground/src/weblets/tf_nextcloud.vue [hunk 1]
```diff
@@ -28,199 +29,222 @@
...
+  layout.value.setStatus(
+    "success",
+    "Successfully deployed a Nextcloud instance. Under Actions, click on the button Nextcloud Setup to set up Nextcloud. After installation, you can access the Nextcloud instance by clicking on the Open Nextcloud button or navigating to your Nextcloud domain.",
+  );
+  layout.value.openDialog(deployment, deploymentListEnvironments.nextcloud);
+}
+async function deploy(gatewayName: GatewayNode, customDomain: boolean) {
   layout.value.setStatus("deploy");
 
   const projectName = ProjectName.Nextcloud.toLowerCase();
```
Comment (MohamedElmdary) [inline packages/playground/src/weblets/tf_nextcloud.vue:140]:
```suggestion
  const projectName = ProjectName.Nextcloud.toLowerCase() + "/" + name.value;
```
Please Add /{name} as suffix to projectName as we changing the way we storing all machines to include gateways

**Comment by Mahmoud-Emad** [comment]
Hello @scottyeager :)

Good job :), I've tried to deploy an instance of `NextCloud` multiple times and I got this error.

image

**Comment by AhmedHanafy725** [comment]
> Good job :), I've tried to deploy an instance of `NextCloud` multiple times and I got this error.
> 
> image

I can see that the name starts with `NaN` which means something undefined is used in the name and the name contract is rejected as it contains uppercase letters

**Comment by scottyeager** [comment]
I've reverted the suggestion that creates the error. @Mahmoud-Emad you can try testing again now :)

@MohamedElmdary, it looks like we need to wait for your PR with the gateway updates to be merged before changing the project name here.

File: packages/playground/src/weblets/tf_deployment_list.vue [hunk 2]
```diff
@@ -187,9 +187,15 @@
           />
           <IconActionBtn
             tooltip="Open Nextcloud"
+            color="white"
```
Comment (Mahmoud-Emad) [inline packages/playground/src/weblets/tf_deployment_list.vue:190]: Please remove the static color here, it should be dynamic

see in dark mode 

- image

light mode 
- image

File: packages/playground/src/weblets/tf_deployment_list.vue [hunk 3]
```diff
@@ -187,9 +187,15 @@
...
             icon="mdi-web"
-            color="anchor"
-            :href="'https://' + item.value[0].env.NEXTCLOUD_DOMAIN + ':8443'"
+            :href="'https://' + item.value[0].env.NEXTCLOUD_DOMAIN"
+          />
+          <IconActionBtn
+            tooltip="Nextcloud Setup"
+            color="white"
```
Comment (Mahmoud-Emad) [inline packages/playground/src/weblets/tf_deployment_list.vue:196]: The same goes here

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]

File: packages/playground/src/weblets/tf_deployment_list.vue (hunk 3 above)
Comment (Mik-TF) [reply packages/playground/src/weblets/tf_deployment_list.vue]: Done. We set "anchor" as in all the other weblets, so it's uniform on the playground.

**Pull request merged**